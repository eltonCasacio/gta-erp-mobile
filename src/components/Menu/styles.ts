import styled from 'styled-components/native';
import Theme from '../../styles/Theme';

export const Wrapper = styled.TouchableOpacity`
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: ${Theme.Spacings.small};
  background-color: ${Theme.Colors.secondary};
`;

export const Text = styled.Text`
  color: ${Theme.Colors.white};
  font-size: ${Theme.Font.small};
  text-align: center;
`;
