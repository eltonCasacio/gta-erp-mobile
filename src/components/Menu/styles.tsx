import styled from 'styled-components/native';
import Theme from '../../styles/Theme';

export const ScrollNavigator = styled.ScrollView`
  height: 120px;
`;

export const NavigatorContent = styled.View``;

export const ViewNavigator = styled.View`
  flex-direction: row;
  margin-left: ${Theme.Spacings.great};
`;

export const NavigatorBox = styled.TouchableOpacity`
  background-color: ${Theme.Colors.secondary};
  height: 100px;
  width: 90px;
  margin-right: ${Theme.Spacings.medium};
  border-radius: ${Theme.Spacings.small};
  padding: ${Theme.Spacings.medium};
  justify-content: space-between;
`;

export const Title = styled.Text`
  color: ${Theme.Colors.white};
`;

export const Icon = styled.Image`
  height: ${({height}) => height};
  width: ${({width}) => width};
`;
