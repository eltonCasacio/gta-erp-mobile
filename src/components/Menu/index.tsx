import React, {useState} from 'react';
import * as S from './styles';

type MenuItemsProps = {
  name: string;
  source: string;
};
const Menu = () => {
  const [menuItems] = useState([
    {name: 'Indicar amigos', source: 'indicator'},
    {name: 'Depositar', source: 'deposit'},
    {name: 'Transferir', source: 'transfer'},
    {name: 'Ajustar Limite', source: 'ajust'},
    {name: 'Cartão virtual', source: 'card'},
    {name: 'Pagar', source: 'payment'},
    {name: 'Bloquear cartão', source: 'block'},
  ]);

  const renderMenusItems = ({name}: MenuItemsProps) => (
    <S.NavigatorBox key={`${name}`}>
      {/* <Icon height={40} width={40} source={source} /> */}
      <S.Title>{name}</S.Title>
    </S.NavigatorBox>
  );

  return (
    <S.NavigatorContent>
      <S.ScrollNavigator horizontal showsHorizontalScrollIndicator={false}>
        {menuItems.map(renderMenusItems)}
      </S.ScrollNavigator>
    </S.NavigatorContent>
  );
};

export default Menu;
