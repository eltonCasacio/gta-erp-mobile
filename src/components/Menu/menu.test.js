import React from 'react';
import renderer from 'react-test-renderer';

import Menu from '.';

describe('<Menu>', () => {
  it('Testing layout/styles', () => {
    const tree = renderer.create(<Menu />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
