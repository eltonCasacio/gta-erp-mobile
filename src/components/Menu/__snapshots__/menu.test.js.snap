// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu> Testing layout/styles 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <RCTScrollView
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    style={
      Array [
        Object {
          "height": 120,
        },
      ]
    }
  >
    <View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5762F2",
            "borderRadius": 5,
            "height": 100,
            "justifyContent": "space-between",
            "marginRight": 10,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
              },
            ]
          }
        >
          Indicar amigos
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5762F2",
            "borderRadius": 5,
            "height": 100,
            "justifyContent": "space-between",
            "marginRight": 10,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
              },
            ]
          }
        >
          Depositar
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5762F2",
            "borderRadius": 5,
            "height": 100,
            "justifyContent": "space-between",
            "marginRight": 10,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
              },
            ]
          }
        >
          Transferir
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5762F2",
            "borderRadius": 5,
            "height": 100,
            "justifyContent": "space-between",
            "marginRight": 10,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
              },
            ]
          }
        >
          Ajustar Limite
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5762F2",
            "borderRadius": 5,
            "height": 100,
            "justifyContent": "space-between",
            "marginRight": 10,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
              },
            ]
          }
        >
          Cartão virtual
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5762F2",
            "borderRadius": 5,
            "height": 100,
            "justifyContent": "space-between",
            "marginRight": 10,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
              },
            ]
          }
        >
          Pagar
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#5762F2",
            "borderRadius": 5,
            "height": 100,
            "justifyContent": "space-between",
            "marginRight": 10,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
            "width": 90,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fafafa",
              },
            ]
          }
        >
          Bloquear cartão
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
